### PR TITLE
Do not recompute the same quantities several times over in the heat sources

### PR DESCRIPTION
**Pull request by Rombur**
By having a function `update_time`, we don't have to recompute everything at each quadrature point.

### CODE REVIEW

**Comment by Rombur** [comment]
Testing a change in the CI

**Comment by Rombur** [comment]
one more test

**Comment by Rombur** [comment]
one last time

**Comment by Rombur** [comment]
OK I have no idea what's the problem with jenkins. The error is 
```
 the credentials Jenkins is using is probably wrong. Or the user account does not have write access to the repo.
```
but Damien is using his credentials on Kokkos without any problem. I've gave him admin permission but nothing works :cry:

File: source/ThermalPhysics.templates.hh [hunk 1]
```diff
@@ -118,6 +118,8 @@ evaluate_thermal_physics_impl(
 
   // Compute the source term.
   // TODO do this on the GPU
+  for (auto &beam : heat_sources)
+    beam->update_time(t);
```
[the remark this answers is not in the document]
Comment (Rombur) [reply source/ThermalPhysics.templates.hh:122]: I wanted to do it that way at first but it makes the code more complicated. The problem is that  on the host version the source is computed by `ThermalOperator` in a matrix-free fashion. For the device, the source is computed in `ThermalPhysics`. When we move the computation of the source on the device and in `ThermalOperatorDevice`, I will move this code into `ThermalOperatorDevice::set_time_and_source_height`

[the remark this answers is not in the document]
Comment (stvdwtt) [reply source/ThermalPhysics.templates.hh:122]: Sounds good -- merging it in.

**Pull request merged**